mesonbuild/dependencies/ui.py

```diff
@@ -378,8 +381,18 @@ def _qmake_detect(self, mods, kwargs):
         libdir = qvars['QT_INSTALL_LIBS']
         # Used by self.compilers_detect()
         self.bindir = self.get_qmake_host_bins(qvars)
+        self.libdir = libdir
+        self.install_prefix = qvars['QT_INSTALL_PREFIX']
         self.is_found = True
 
+        qconfig_path = os.path.join(qvars['QT_INSTALL_PREFIX'], 'mkspecs', 'qconfig.pri')
```

Comment:
```suggestion
        qconfig_path = os.path.join(qvars['QT_INSTALL_ARCHDATA'], 'mkspecs', 'qconfig.pri')
```
To my understanding https://doc.qt.io/qt-6/qmake-environment-reference.html says that mkspecs is in the data dir and because it's architecture-dependent QT_INSTALL_ARCHDATA should be used.